src/components/table/z-td/index.tsx

```diff
@@ -0,0 +1,97 @@
+import {Component, Element, Event, EventEmitter, Host, Prop, State, Watch, h} from "@stencil/core";
+import {ButtonVariant, ControlSize} from "../../../beans";
+
+/**
+ * ZTd component.
+ * @slot - ZTd content.
+ */
+@Component({
+  tag: "z-td",
+  styleUrl: "styles.css",
+  shadow: true,
+})
+export class ZTd {
+  @Element() host: HTMLZTdElement;
+
+  /**
+   * Number of columns that the cell should span.
+   */
+  @Prop()
+  colspan: number;
+
+  /**
+   * Whether the cell should stick.
+   */
+  @Prop({reflect: true})
+  sticky = false;
+
+  /**
+   * Enables the contextual menu.
+   * If true, a contextual menu button will be shown on hover.
+   */
+  @Prop({reflect: true})
+  showMenu = false;
+
+  /**
+   * Store the open state of the menu.
+   */
+  @State()
+  isMenuOpen = false;
+
+  private menuTrigger: HTMLZButtonElement;
+
+  private popoverEl: HTMLZPopoverElement;
+
+  /**
+   * Emitted when the value of the `colspan` changes.
+   */
+  @Event({
+    bubbles: true,
+    cancelable: false,
+  })
+  colspanChange: EventEmitter<number>;
+
+  @Watch("colspan")
+  protected updateColspan(): void {
+    if (this.colspan) {
+      this.host.style.gridColumn = `span ${this.colspan}`;
+    } else {
+      this.host.style.removeProperty("grid-column");
+    }
+
+    this.colspanChange.emit(this.colspan || 1);
+  }
+
+  componentWillLoad(): void {
+    this.updateColspan();
+  }
+
+  render(): HTMLZTdElement {
+    return (
+      <Host
+        role="cell"
+        menu-open={this.isMenuOpen}
+      >
+        <slot></slot>
+        {this.showMenu && (
+          <div class="z-td--menu-container prevent-expand">
+            <z-button
+              variant={ButtonVariant.TERTIARY}
+              icon="contextual-menu"
+              size={ControlSize.X_SMALL}
+              ref={(el) => (this.menuTrigger = el as HTMLZButtonElement)}
+              onClick={() => (this.popoverEl.open = !this.popoverEl.open)}
+            />
+            <z-popover
+              ref={(el) => (this.popoverEl = el as HTMLZPopoverElement)}
+              bindTo={this.menuTrigger}
+              onOpenChange={(event) => (this.isMenuOpen = event.detail.open)}
+            >
+              <slot name="contextual-menu"></slot>
```

Comment:
Il componente contextual-menu attualmente a DS incorpora anche il button di apertura/chiusura, quindi attualmente il prodotto dovrebbe ricrearsi il menu per inserirlo in questa tabella. A me questa implementazione torna, a questo punto però il contextual menu va buttato :wastebasket: :smile: o comunque rivisto per eventualmente strapolarlo dal trigger.

Reply:
sono d'accordo che va rivisto. tutte le volte in cui volevo usarlo, non soddisfaceva le richieste di design (tipo in questo caso non si poteva fare un button tertiary) 😬 magari possiamo discutere anche di questo con gli altri